### PR TITLE
feat(in-app-messaging): add NextJS example apps

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add In-App Messaging NextJS example apps:
- basic usage with `InAppMessagingProvider`
- basic usage with `withInAppMessaging`
- component override
- update style override

#### Issue #, if available
NA

#### Description of how you validated changes
Manually tested sample apps
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+const CustomBannerMessage = (props) => {
+  const theme = useTheme();
+
+  return (
+    <Flex
+      alignItems="center"
+      borderRadius=".5rem"
+      position="absolute"
+      padding="2rem"
+      backgroundColor={theme.tokens.colors.teal[20]}
```
Comment (zchenwei) [inline examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx:28]:
```suggestion
      backgroundColor={teal.20}
```

So we do not need `useTheme();`

File: examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+const CustomBannerMessage = (props) => {
+  const theme = useTheme();
+
+  return (
+    <Flex
+      alignItems="center"
+      borderRadius=".5rem"
+      position="absolute"
+      padding="2rem"
```
Comment (zchenwei) [inline examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx:27]:
```suggestion
      padding="xl"
```

File: examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      backgroundColor={theme.tokens.colors.teal[20]}
+      right="2rem"
```
Comment (zchenwei) [inline examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx:29]:
```suggestion
      right="xl"
```

File: examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Amplify.configure(config);
+
+const CustomBannerMessage = (props) => {
+  const theme = useTheme();
+
+  return (
+    <Flex
+      alignItems="center"
+      borderRadius=".5rem"
```
Comment (zchenwei) [inline examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx:25]:
```suggestion
      borderRadius="xs"
```

File: examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useEffect } from 'react';
+import { Amplify, Notifications } from 'aws-amplify';
+import {
+  Button,
+  Flex,
+  Text,
+  useInAppMessaging,
+  useTheme,
```
Comment (calebpollman) [inline examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx:8]:
```suggestion
```

File: examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+import config from './aws-exports';
+
+const { InAppMessaging } = Notifications;
+
+Amplify.configure(config);
+
+const CustomBannerMessage = (props) => {
+  const theme = useTheme();
+
```
Comment (calebpollman) [inline examples/next/pages/ui/components/in-app-messaging/override-component/index.page.tsx:21]:
```suggestion
```

**Review by zchenwei — APPROVED** [review]
LGTM

**Review by joebuono — APPROVED** [review]

**Pull request merged**